frameworks/rooch-framework/sources/upgrade.move

```diff
@@ -2,28 +2,77 @@
 // SPDX-License-Identifier: Apache-2.0
 
 module rooch_framework::upgrade {
+    use std::vector;
 
     use moveos_std::gas_schedule::update_gas_schedule;
     use moveos_std::signer::module_signer;
     use moveos_std::signer;
     use moveos_std::event;
-    
+    use moveos_std::bcs;
     use moveos_std::account::create_signer_for_system;
     use moveos_std::module_store;
+    use moveos_std::object;
     use rooch_framework::onchain_config;
 
+    friend rooch_framework::genesis;
+
     const ErrorNotSequencer: u64 = 1;
+    const ErrorCapabilityAlreadyExists: u64 = 2;
+    const ErrorNoAccess: u64 = 3;
 
     const MoveStdAccount: address = @0x1;
     const MoveosStdAccount: address = @0x2;
     const RoochFrameworkAccount: address = @rooch_framework;
     const BitcoinMoveAccount: address = @0x4;
 
+    #[data_struct]
+    struct StdlibPackage has store, copy, drop {
+        package_name: std::string::String,
+        genesis_account: address,
+        modules: vector<vector<u8>>,
+    }
+
+    #[data_struct]
+    /// Collection of framework packages. The struct must keep the same with the Rust definition.
+    struct Stdlib has store, copy, drop {
+        packages: vector<StdlibPackage>,
+    }
+
+    /// Upgrade capability
+    struct UpgradeCap has key, store {}
```

Comment:
Should we move the UpgradeCap to move_module, and create a UpgradeCap for every Package a?